src/Nethermind/Nethermind.Synchronization/FastBlocks/FastBlockStatusList.cs

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Nethermind.Synchronization.Test")]
+namespace Nethermind.Synchronization.FastBlocks;
+
+internal class FastBlockStatusList
```

Comment:
Hm... Are there any generic solutions to pack small enum values collections into bytes? This looks like a generalizable problem, maybe even in our codebase.

Reply:
Would need to work out the max value so how much could be packed 🤔

Could do something like:
```csharp
class PackedEnumList<TEnum>
    where TEnum : struct, Enum
{
    public PackedEnumList(long length)
    {
        // Allocatey
        TEnum[] values = Enum.GetValues<TEnum>();
        // Need to get max value here and convert to int even though TEnum is not numeric
```
Would need to find the underlying type `byte`, `short`, `int` etc could cache that across invocations; isn't completely straight forward.

Will raise discussion issue in runtime to see if the new INumeric style interfaces can be applied to enums; which would make it easier

Reply:
We are using `FastEnum` package for optimizations.

Reply:
Yeah for now it would have to be explicit like:

```
    class PackedEnumList<TEnum, TUnderlyingType>
        where TEnum : struct, Enum
        where TUnderlyingType : INumber<TUnderlyingType>
```

Reply:
> We are using `FastEnum` package for optimizations.

Can do `TEnum? max = FastEnum.GetMaxValue<TEnum>()`, but that returns the enum type which isn't easily convertible to numeric to workout the power of two and the compressibility. Something could definitely be done though, would be easier with a second or third example of arrays of enums to generalize over (rather then necessarily create a complete package)

Reply:
Starts to look complicated, let's explore if needed elsewhere. But potentially could be neat little library.

```cs
    class PackedEnumList<TEnum, TUnderlyingType>
        where TEnum : struct, Enum
        where TUnderlyingType : IBinaryInteger<TUnderlyingType>
    {
        public PackedEnumList()
        {
            if (typeof(TUnderlyingType) != FastEnum.GetUnderlyingType<TEnum>())
                throw new InvalidOperationException($"Underlying type of {nameof(TEnum)} must be the same as {nameof(TUnderlyingType)}");

            TEnum? maxValue = FastEnum.GetMaxValue<TEnum>();
            if (maxValue is null) throw new InvalidOperationException($"Type of {nameof(TEnum)} must have max value");
            TEnum maxValueValue = maxValue.Value;

            TUnderlyingType maxUnderlyingType = Unsafe.As<TEnum, TUnderlyingType>(ref maxValueValue);
            int mask = 1;
            while (maxUnderlyingType > TUnderlyingType.One)
            {
                maxUnderlyingType >>= 1;
                mask = (mask << 1) | 1;
            }
        }
    }
```